src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.CoreRT.cs

```diff
@@ -0,0 +1,476 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using Internal.Runtime.CompilerServices;
+
+namespace System.Runtime.InteropServices
+{
+    /// <summary>
+    /// Class for managing wrappers of COM IUnknown types.
+    /// </summary>
+    public abstract partial class ComWrappers
+    {
+        internal static IntPtr DefaultIUnknownVftblPtr { get; } = CreateDefaultIUnknownVftbl();
+
+        internal static Guid IID_IUnknown = new Guid(0x00000000, 0x0000, 0x0000, 0xc0, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x46);
+
+        private static readonly ConditionalWeakTable<object, ManagedObjectWrapperHolder> CCWTable = new ConditionalWeakTable<object, ManagedObjectWrapperHolder>();
+
+        /// <summary>
+        /// ABI for function dispatch of a COM interface.
+        /// </summary>
+        public unsafe partial struct ComInterfaceDispatch
+        {
+            /// <summary>
+            /// Given a <see cref="System.IntPtr"/> from a generated Vtable, convert to the target type.
+            /// </summary>
+            /// <typeparam name="T">Desired type.</typeparam>
+            /// <param name="dispatchPtr">Pointer supplied to Vtable function entry.</param>
+            /// <returns>Instance of type associated with dispatched function call.</returns>
+            public static unsafe T GetInstance<T>(ComInterfaceDispatch* dispatchPtr) where T : class
+            {
+                ManagedObjectWrapper* comInstance = ToManagedObjectWrapper(dispatchPtr);
+                return Unsafe.As<T>(RuntimeImports.RhHandleGet(comInstance->Target));
+            }
+
+            internal static unsafe ManagedObjectWrapper* ToManagedObjectWrapper(ComInterfaceDispatch* dispatchPtr)
+            {
+                return ((InternalComInterfaceDispatch*)dispatchPtr)->thisPtr;
+            }
+        }
+
+        internal unsafe struct InternalComInterfaceDispatch
+        {
+            public IntPtr Vtable;
+            internal ManagedObjectWrapper* thisPtr;
+        }
+
+        internal enum CreateComInterfaceFlagsEx
+        {
+            None = CreateComInterfaceFlags.None,
+            CallerDefinedIUnknown = CreateComInterfaceFlags.CallerDefinedIUnknown,
+            TrackerSupport = CreateComInterfaceFlags.TrackerSupport,
+
+            // Highest bits are reserved for internal usage
+            LacksICustomQueryInterface = 1 << 29,
+            IsComActivated = 1 << 30,
+            IsPegged = 1 << 31,
+
+            InternalMask = IsPegged | IsComActivated | LacksICustomQueryInterface,
```

Comment:
```suggestion
            // Highest bits are reserved for internal usage
            LacksICustomQueryInterface = 1 << 29,
            IsComActivated = 1 << 30,
            IsPegged = 1 << 31,

            InternalMask = IsPegged | IsComActivated | LacksICustomQueryInterface,
```
None of this is used in this PR. It would be nice to omit `CreateComInterfaceFlagsEx` from this PR and only introduce later once it is actually needed.

Reply:
Will remove